workshops/socialauth.py

```diff
@@ -0,0 +1,76 @@
+from django.http import HttpResponseForbidden
+
+from django.http import HttpResponse
+
+from selectable.decorators import results_decorator
+
+from django.contrib.auth.mixins import UserPassesTestMixin
+
+from django.contrib.auth.decorators import user_passes_test
+
+from django.contrib.auth.decorators import login_required
+
+from django.core.urlresolvers import reverse
+
+from django.shortcuts import redirect, render
+
+from workshops.models import Person
+
+
+class NoUserAssociatedWithGithubAccount(Exception):
+    pass
+
+
+def abort_if_no_user(user, *args, **kwargs):
+    if user is None:
+        raise NoUserAssociatedWithGithubAccount
+
+
+class NoUserAssociatedWithGithubAccountMiddleware(object):
+    def process_exception(self, request, exception):
+        if isinstance(exception, NoUserAssociatedWithGithubAccount):
+            return redirect(reverse('no_github_user'))
+
+
+def find_user_or_abort(details=None, **kwargs):
+    username = details['username']
+    try:
+        user = Person.objects.get(github=username)
+        return {'user': user}
+    except Person.DoesNotExist:
+        raise NoUserAssociatedWithGithubAccount
```

Comment:
No, it's equivalent to `raise NoUserAssociatedWithGithubAccount()`. See http://stackoverflow.com/questions/16706956/is-there-a-difference-between-raise-exception-and-raise-exception-without